test/cql-pytest/test_describe.py

```diff
@@ -0,0 +1,344 @@
+# Copyright 2022-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+#############################################################################
+# Tests for server-side describe
+#############################################################################
+
+# Tests for listing describe statements compares the result from describe to
+# the corresponding table from `system_schema` keyspace. This way, the tests ensure
+# all of the elements are present in the describe listing.
+#
+# To test correctness of create statement in element describe 
+# (e.g. `DESC TABLE ks.tbl`), test creates a random element using functions from
+# `random_util`. 
+# Then, the new element is recreated using statement from describe (with different name).
+# The test compares old and new elements by comparing theris rows from 
+# appropriate table in `system_schema` keyspace.
+
+import pytest
+from random_util import new_random_table, new_random_type, new_random_keyspace
+from util import new_test_table, new_type, new_function, new_aggregate, new_session, unique_name
+
+def get_name(name_with_ks):
+    return name_with_ks.split(".")[1]
+
+# Test `DESC {elements}` listing. Assert if all rows from `system_schema.{elements}`
+# are in describe result
+def assert_element_listing(cql, elements, name_column, name_f):
+    # Crate new session to ...
+    with new_session(cql, "cassandra") as ncql:
+        table_result = ncql.execute(f"SELECT keyspace_name, {name_column} FROM system_schema.{elements}")
+        desc_result = ncql.execute(f"DESC {elements}")
+
+        desc_listing = [(r.keyspace_name, r.name) for r in desc_result]
+
+        for row in table_result:
+            assert (row.keyspace_name, name_f(row)) in desc_listing
+
+# Test that `DESC KEYSPACES` contains all keyspaces
+def test_keyspaces(cql):
+    assert_element_listing(cql, "keyspaces", "keyspace_name", lambda r: r.keyspace_name)
```

Comment:
I think your test should make sure there are some user keyspaces before it tries to list them. Maybe just add the "test_keyspace" fixture to this test, to just ensure at least this one keyspace exists.

Reply:
added